Fixes for issue #110

bluetiger9 opened the pull request:
Fixes `exclusive_caps=1` related problems, from the [issue #110](https://github.com/umlaeute/v4l2loopback/issues/110):

The first commit fixes problems caused by the incorrectly set variables `dev->ready_for_output` and `opener->type`. These were also incorrectly set on `try/get*` calls, causing `Video Output` capability to be announced to early.

The seconds commit is a workaround for `GStreamer` failure on enumerating inputs. Seems that `GStreamer` tries to enumerate the inputs even if `Video Capture` capability is not announced.

umlaeute commented:
i'm very reluctant to accept any pull-requests for gstreamer-0.10 fixes.

*GStreamer-0.10* has been dead or a while now, and should be replaced by *GStreamer-1.0*.

the entire `exclusive_caps` foo is a mess and only needed because some clients (GStreamer-0.10, Google Chrome) decided to do their very own thing incompatible with the rest (and incompatible with each other). i'd rather have it go away completely.

so you need very good persuasive powers to make me accept a change the sole purpose of which is to revert ca8639cd which was only introduced for the same reason (#65)

bluetiger9 commented:
Regarding to `exclusive_caps`. Yes, it's a feature present to comply with some applications, but I think is pretty clear how should work. And with the proposed changes I think it works as expected (forget about gstreamer for now). 

I don't think it is good idea to remove features, just to improve code quality or because they are hackish. You can remove it, but from the user's point of view, not only that there is no benefit, but (as the compatibility with highly used applications like Google Chrome will be broken) the library will be seen worse that the previous version.

Regarding to the `GStreamer` bug, the correct solution in theory is clearly the old one. But, as long the client application ask stupid/invalid questions, I think is it's fine to return stupid answers if that is needed for compatibility. It's basically a zero risk change. I don't think there is an application that intentionally ask stupid questions and than fails if no error was returned. (note: the change has nothing to do with bug fixed in ca8639c, I removed the entire if, so it should be fine)

PS: There is something wrong with ```GStreamer-1.0``` on the computer I tried. First tried with it, but the output was laggy as hell, so I switched to ```GStreamer-0.1```.

chikamichi commented:
Hi,

Any thoughts on that one PR, @umlaeute?

umlaeute commented:
merged
